### PR TITLE
Add user_authentication to delete and edit restaurants

### PR DESCRIPTION
**Pull request by atasker**

### CODE REVIEW

**Comment by omidbachari** [comment]
This looks pretty short and sweet. Merge when ready @atasker

File: app/models/restaurant.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Restaurant < ActiveRecord::Base
-  has_many :reviews, :dependent => :destroy
+  has_many :reviews
```
Comment (dpickett) [inline app/models/restaurant.rb:2]: is a review relevant when a restaurant is destroyed? I think the destroy dependency is appropriate here.

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,5 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  has_many :restaurants
```
Comment (dpickett) [inline app/models/user.rb:6]: we should probably nullify the restaurants author if we destroy the user. checkout the `:nullify` option for when you define a destroy dependency

File: db/migrate/20150331143401_add_admin_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddAdminToUsers < ActiveRecord::Migration
+  def change
+    add_column :users, :admin, :boolean, default: false
```
Comment (dpickett) [inline db/migrate/20150331143401_add_admin_to_users.rb:3]: it is likely that you will need to define more than just an admin role. In general, I would suggest defining a `role` string column that is restricted to certain values. Your migration might look like this:

``` ruby
add_column :users, :role, :string, default: 'user'
```

**Pull request merged**